edc-extensions/migrations/control-plane-migration/src/main/resources/org/eclipse/tractusx/edc/postgresql/migration/contractnegotiation/V0_0_10__Init_ContractAgreementRetirement.sql

```diff
@@ -0,0 +1,37 @@
+--
+--  Copyright (c) 2024 Mercedes-Benz Tech Innovation GmbH
+--
+--  This program and the accompanying materials are made available under the
+--  terms of the Apache License, Version 2.0 which is available at
+--  https://www.apache.org/licenses/LICENSE-2.0
+--
+--  SPDX-License-Identifier: Apache-2.0
+--
+--  Contributors:
+--       Mercedes-Benz Tech Innovation GmbH - Initial Database Schema
+--
+
+--
+-- table: edc_contract_agreement
+--
+CREATE TABLE IF NOT EXISTS edc_contract_agreement
```

Comment:
Also i would not put this migration under the `contractnegotiation` sub migration but rather in it's own folder

and as @ndr-brt  i would not use the foreign key constraint

Reply:
Was unaware on how to reference from other file, tbh. Either way, removed in [here](https://github.com/eclipse-tractusx/tractusx-edc/pull/1631/commits/223e47d48cb7fdc2807da6ecb9589541a30d208d) due to not having to reference this table for the fk.
Decided to query the contract_agreement db, however this made me replicate table and column names of that table in the retirement store, which duplicates this info. This is due to the default access modifier on these methods. Tried to find similar scenarios, but was not able to.

Reply:
> Also i would not put this migration under the `contractnegotiation` sub migration since it's a different datasource but rather in it's own folder
> 
> and as @ndr-brt i would not use the foreign key constraint

I thought leaving it at the same level of contract_agreement would make more sense since this new table exists almost as an "extension" of the previous one

Reply:
@wolf4ood changed [here.](https://github.com/eclipse-tractusx/tractusx-edc/pull/1631/commits/ff1148784d0721fd5017d90eba109891a97cdd1f)